Kudosbot -- initial working version

ericzhang98 opened the pull request:
Initial working version of Kudosbot:
- the bot logic lives in `KudosbotService` which is largely copied from `TelegramService` and uses the same `constructProofUrl` construction as the demo client
- the `kudos` command calls the proof construction logic and uploads the proof (via redirect url) to the aggregation server
- the `refresh` commands populates the `telegram_bot_conversations` table with usernames
- add https support for KudosBot demo client and upload / aggregation server
- add `KUDOSBOT_TELEGRAM_TOKEN` env variable for local bot testing
- pulls in the db migration to add a `telegram_username` column from https://github.com/proofcarryingdata/zupass/pull/993

tested the `kudos` and `reset` command locally with a copy of the staging db

rrrliu commented on apps/kudosbot-server/src/routing/routes/uploadRoute.ts:
```diff
@@ -12,6 +12,14 @@ export function initUploadRoute(
 ): void {
   console.log("[INIT] Initializing upload route");
 
+  const deserializeKudosData = (kudosData: string) => {
```
all the logic in this file should go within a kudosRoutes.ts file in passport-server

rrrliu commented on apps/passport-server/src/services/kudosbotService.ts:
```diff
@@ -0,0 +1,305 @@
+const ZUPASS_CLIENT_URL =
+  process.env.PASSPORT_CLIENT_URL || "https://dev.local:3000";
+// TODO: figure out url for upload / aggregation server
+const KUDOSBOT_SERVER_UPLOAD_URL = "https://dev.local:3005/upload";
```
this should be replaced with a passport-server route

rrrliu commented on apps/passport-server/src/services/kudosbotService.ts:
```diff
@@ -0,0 +1,305 @@
+      await ctx.reply("hello");
+    });
+
+    const getSemaphoreIdFromTelegramUsername = async (
```
this should go in under `queries/telegram/`

rrrliu commented on apps/passport-server/src/services/kudosbotService.ts:
```diff
@@ -0,0 +1,305 @@
+      return semaphoreId;
+    };
+
+    this.bot.command("test", async (ctx) => {
```
do we need this test command?

rrrliu commented on apps/passport-server/src/services/telegramService.ts:
```diff
@@ -59,7 +59,8 @@ const ALLOWED_TICKET_MANAGERS = [
   "gubsheep",
-  "chubivan"
+  "chubivan",
+  "ezboi98"
```
this is fine for testing, but don't commit this

ericzhang98 commented:
@rrrliu addressed all your comments:
- removed unused test stuff and moved db query to separate file
- changed kudosbot telegram token env var to TELEGRAM_KUDOSBOT_TOKEN
- removed `kudosbot-server` and moved old upload logic into `/kudos/upload` route in `passport-server` (in a kudosbot routes file)
- updated url constants to use passport server url

note that the old `kudosbot-client` demo code is still in the repo and I'll be expanding on that when building the frontend client for viewing kudos aggregate info

rrrliu reviewed (commented):
nice! just one note, and should be ready to go

ericzhang98 commented:
@rrrliu updated to env var
